main.c

```diff
@@ -1,85 +1,126 @@
 #include <sys/ioctl.h>
-#include <stdio.h>
 #include <unistd.h>
 #include <pthread.h>
-
-
-
-void clearScreen() {
-    printf("\033[H\033[J");
-}
-void gotoxy(int x, int y) {
-    printf("\033[%d;%dH", (y), (x));
-}
-
+#include <stdio.h>
+#include <ctype.h>
+#include "ascii_symbols.h"
+#include "term_commands.h"
+#include "tv100_commands.h"
 
 struct winsize w;
 
 int stop = 0;
 
-void clearLine () {
-    printf("%c[2K", 27);
-    printf("\r");
-    fflush(stdout);
-}
-
-void printString (char* str) {
-    printf("%s", str);
-    fflush(stdout);
-}
 
-void saveCursorPos () {
-    printf("%c7", 27);
-    fflush(stdout);
+void printString (char* str)
+{
+  printf("%s", str);
+  fflush(stdout);
 }
 
-void restoreCursorPos () {
-    printf("%c8", 27);
-    fflush(stdout);
-}
 
-void *drawProgress(void *null)
+void redrawPrevLines (char matrix[1000][1000], int currentLine)
 {
-    int width = w.ws_col;
-    int height = w.ws_row;
-    int n = 0;
-    while (stop == 0) {
-        usleep(20000);
-        saveCursorPos();
-        gotoxy(n, height);
-        n++;
-        if (n > width - 1) {
-            n = 0;
-            clearLine();
-        }
-        printString("#");
-        restoreCursorPos();
+  const int width = w.ws_col;
+  saveCursorPos();
+  gotoxy(0, 0);
+  clearLine();
+  for (int i = 1; i <= currentLine; ++i) {
+    gotoxy(0, i);
+    clearLine();
+    for (int j = 1; j <= width; ++j) {
+      gotoxy(j, i);
+      printf("%c", matrix[i][j]);
     }
-    return NULL;
-
+  }
+  restoreCursorPos();
 }
 
-int main () {
-    pthread_t drawProgress_thread;
-    int c;
-
-    clearScreen();
-
-    ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
-    if(pthread_create(&drawProgress_thread, NULL, drawProgress, NULL)) {
-
-        fprintf(stderr, "Error creating thread\n");
-        return 1;
-
+void *drawProgress (void *null)
+{
+  const int width = w.ws_col;
+  int currentRow = 0;
+  while (stop == 0) {
+    usleep(20000);
+    saveCursorPos();
+    // Moves carret one line up and to start of line
+    printSpecialChar("[1A\r");
+    currentRow++;
+    if (currentRow > width - 1) {
+      currentRow = 0;
+      clearLine();
     }
-    if(pthread_join(drawProgress_thread, NULL)) {
-
-        fprintf(stderr, "Error joining thread\n");
-        return 2;
+    for (int i = 0; i < currentRow; ++i) {
+      printString("#");
+    }
+    restoreCursorPos();
+  }
+  return NULL;
+}
 
+void *proceedInput (void *null)
+{
+  int c;
+  // Matrix shpuld be limited to width and height
+  // But unfortunantely for now it will be 1000 * 1000
+  char charMatrix[1000][1000];
+  int currentLine = 1;
+  int currentRow = 1;
+
+  int quitSeqIter = 0;
+  while((c = getchar()) != EOF) {
+    printf("%c", c);
+    if (tolower(c) == QUIT[quitSeqIter] || c == ESC || c == CANCEL) {
+      if (quitSeqIter == 3 || c == ESC || c == CANCEL) {
+        stop = 1;
+        printf("\n\r");
+        return NULL;
+      }
+      quitSeqIter++;
```

Comment:
Please don't use postfix increments(which make a copy) where you don't really need it.